### ISSUE TITLE
Stress test investigation - 404 GetLastMiniblockHash

### ISSUE THREAD

**Issue by texuf** [river-build/river#643]
### Description

What's going on here? Can we add more logs and try to figure out if this is a stream that actually doesn't exist, or if this is a bug on the server

link to logs: https://app.datadoghq.com/logs?query=env%3Agamma%20service%3Astress-test-node%20&agg_m=count&agg_m_source=base&agg_t=count&cols=env%2Cservice%2Ccontainer_index&event=AgAAAZEpTNRihrB34QAAAAAAAAAYAAAAAEFaRXBUTjRnQUFDU1hfRFF6VkQxREFCYQAAACQAAAAAMDE5MTI5NGMtZWFhNy00YjYzLTljMDEtMDY4ZDYwNjZhNDc3&fromUser=true&messageDisplay=inline&refresh_mode=sliding&saved-view-id=2705025&storage=hot&stream_sort=desc&view=spans&viz=stream&from_ts=1722876416029&to_ts=1722880016029&live=true

example logs

```
2024-08-06T20:06:26.657Z csb:rpc:error GetLastMiniblockHash ERROR kKh2HszfdiRH _ConnectError: [unimplemented] HTTP 404 at validateResponse (/monorepo/packages/stress/dist/start.cjs:156154:29) at validateResponseWithCompression (/monorepo/packages/stress/dist/start.cjs:156171:41) at next (/monorepo/packages/stress/dist/start.cjs:156383:61) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async /monorepo/packages/stress/dist/start.cjs:156782:16 at async /monorepo/packages/stress/dist/start.cjs:156857:19 at async Object.unary (/monorepo/packages/stress/dist/start.cjs:156346:14) at async Object.getLastMiniblockHash (/monorepo/packages/stress/dist/start.cjs:155585:22) at async Client.getStreamLastMiniblockHash (/monorepo/packages/stress/dist/start.cjs:162798:15) at async /monorepo/packages/stress/dist/start.cjs:162903:31 { rawMessage: 'HTTP 404', code: 12, metadata: Headers { 'content-type': 'text/plain; charset=utf-8', vary: 'Origin', 'x-content-type-options': 'nosniff', 'content-length': '19', date: 'Tue, 06 Aug 2024 20:06:26 GMT' }, details: [], cause: undefined }
```

**Comment by texuf** [comment]
Looks like this is still happening, can you please update this task with everything we know and we will pass it on to serge https://app.datadoghq.com/logs?query=env%3Agamma%20service%3Astress-test-node%20404%20&agg_m=count&agg_m_source=base&agg_t=count&cols=env%2Cservice%2Ccontainer_index&fromUser=true&messageDisplay=inline&refresh_mode=sliding&saved-view-id=2705025&storage=hot&stream_sort=desc&view=spans&viz=stream&from_ts=1722537455806&to_ts=1723142255806&live=true

**Comment by texuf** [comment]
I did some investigations locally. I increased the recency constraints to keep many past miniblocks in memory. The error changed from "not found" to "not recent" confirming that the block was truly older than 12 seconds. This is a symptom of another problem also affecting the notifications service and potentially our web clients where the sync backs up and doesn't send the latest messages to all clients.
